lib/msf/core/auxiliary/vyos.rb

```diff
@@ -0,0 +1,312 @@
+# -*- coding: binary -*-
+
+require 'metasploit/framework/hashes/identify'
+
+module Msf
+  ###
+  #
+  # This module provides methods for working with VyOS equipment
+  #
+  ###
+  module Auxiliary::VYOS
+    include Msf::Auxiliary::Report
+
+    def vyos_config_eater(thost, tport, config, store = true)
+
+      credential_data = {
+        address: thost,
+        port: tport,
+        protocol: 'tcp',
+        workspace_id: myworkspace.id,
```

Comment:
I don't remember what best practice is here, but using `myworkspace.id` caused an issue when a database was not connected.

```
msf6 post(networking/gather/enum_vyos) > use auxiliary/scanner/ssh/ssh_login
msf6 auxiliary(scanner/ssh/ssh_login) > set username vyos
username => vyos
msf6 auxiliary(scanner/ssh/ssh_login) > set password vyos
password => vyos
msf6 auxiliary(scanner/ssh/ssh_login) > set rhosts 172.16.191.158
rhosts => 172.16.191.158
msf6 auxiliary(scanner/ssh/ssh_login) > run

[+] 172.16.191.158:22 - Success: 'vyos:vyos' 'uid=1000(vyos) gid=100(users) groups=100(users),4(adm),6(disk),27(sudo),30(dip),102(quaggavty),104(vyattacfg),110(fuse) Linux vyos 3.13.11-1-amd64-vyos #1 SMP Sat Nov 11 12:10:30 CET 2017 x86_64 GNU/Linux '
[*] Command shell session 2 opened (172.16.191.165:34325 -> 172.16.191.158:22) at 2020-09-25 17:43:45 -0400
[*] Scanned 1 of 1 hosts (100% complete)
[*] Auxiliary module execution completed
msf6 auxiliary(scanner/ssh/ssh_login) > use post/networking/gather/enum_vyos 
msf6 post(networking/gather/enum_vyos) > set session 2
session => 2
msf6 post(networking/gather/enum_vyos) > run

[!] SESSION may not be compatible with this module.
[*] Getting version information
[*] Gathering info from cat /config/config
[-] Post failed: NoMethodError undefined method `id' for nil:NilClass
[-] Call stack:
[-]   /root/Desktop/metasploit-framework/lib/msf/core/auxiliary/vyos.rb:20:in `vyos_config_eater'
[-]   /root/Desktop/metasploit-framework/modules/post/networking/gather/enum_vyos.rb:74:in `block in enum_configs'
[-]   /root/Desktop/metasploit-framework/modules/post/networking/gather/enum_vyos.rb:70:in `each'
[-]   /root/Desktop/metasploit-framework/modules/post/networking/gather/enum_vyos.rb:70:in `enum_configs'
[-]   /root/Desktop/metasploit-framework/modules/post/networking/gather/enum_vyos.rb:51:in `run'
[*] Post module execution completed
msf6 post(networking/gather/enum_vyos) > [*] You have active sessions open, to exit anyway type "exit -y"
msf6 post(networking/gather/enum_vyos) > edit /root/Desktop/metasploit-framework/lib/msf/core/auxiliary/vyos.rb
[-] Could not execute /usr/bin/vim /root/Desktop/metasploit-framework/lib/msf/core/auxiliary/vyos.rb
msf6 post(networking/gather/enum_vyos) > edit /root/Desktop/metasploit-framework/lib/msf/core/auxiliary/vyos.rb
[*] Reloading /root/Desktop/metasploit-framework/lib/msf/core/auxiliary/vyos.rb
msf6 post(networking/gather/enum_vyos) > rexploit 
[*] Reloading module...

[!] SESSION may not be compatible with this module.
[*] Getting version information
[*] Gathering info from cat /config/config
[*] Gathering info from cat /config/config.boot
[+] 172.16.191.158:22 Username 'jsmith' with level 'operator' with hash $6$UMf4hTrJ87jq2J$uomBCF8RdynAlb.UeylW9N80mF0GzftDjybTqZmFVNpQCjWDf7aTrVcPUPoJ8z7Ij2.Ee1M84l4r/qiescbyz0
[+] 172.16.191.158:22 Username 'vyos' with level 'admin' with hash $1$j4H9kcPE$RpI7FAxfnj9Ogjn.UhQ0C0
[+] 172.16.191.158:22 Hostname: vyos
[+] 172.16.191.158:22 OS Version: VyOS 1.1.8
[+] 172.16.191.158:22 Interface eth0 (00:0c:29:c7:af:bc)
[*] Post module execution completed
msf6 post(networking/gather/enum_vyos) > 
```

The issue was resolved with the following diff:

```suggestion
        workspace_id: myworkspace_id,
```

The framework appears to use a mix of both `myworkspace_id` and `myworkspace.id` approaches:

```
# grep -rn myworkspace.id lib/msf/core
lib/msf/core/module/auth.rb:12:        workspace_id: myworkspace_id
lib/msf/core/module/external.rb:142:      workspace_id: myworkspace_id
lib/msf/core/rpc/v10/rpc_db.rb:183:  #   workspace_id: myworkspace_id,
lib/msf/core/auxiliary/brocade.rb:58:          workspace_id: myworkspace.id,
lib/msf/core/auxiliary/report.rb:87:  def myworkspace_id
lib/msf/core/auxiliary/report.rb:89:      myworkspace.id
lib/msf/core/auxiliary/report.rb:248:      workspace_id: myworkspace_id
lib/msf/core/auxiliary/ubiquiti.rb:105:          workspace_id: myworkspace.id,
lib/msf/core/auxiliary/cnpilot.rb:21:      workspace_id: myworkspace_id
lib/msf/core/auxiliary/f5.rb:20:        workspace_id: myworkspace.id,
lib/msf/core/auxiliary/epmp.rb:21:      workspace_id: myworkspace_id
lib/msf/core/auxiliary/juniper.rb:34:        workspace_id: myworkspace.id,
lib/msf/core/auxiliary/juniper.rb:158:        workspace_id: myworkspace.id,
lib/msf/core/auxiliary/mikrotik.rb:86:          workspace_id: myworkspace.id,
lib/msf/core/auxiliary/mikrotik.rb:181:          workspace_id: myworkspace.id,
lib/msf/core/auxiliary/cisco.rb:43:          workspace_id: myworkspace.id,
lib/msf/core/auxiliary/vyos.rb:20:        workspace_id: myworkspace_id,
lib/msf/core/auxiliary/arista.rb:21:          workspace_id: myworkspace.id,
lib/msf/core/db_export.rb:151:    @creds = Metasploit::Credential::Core.with_logins.with_public.with_private.workspace_id(myworkspace.id)
```

Reply:
I remember that being a thing now, I *thought* I had resolved it for copy pasta purposes.  Looks like not.
Also looks like a follow on PR to fix all this up.